promql/parser/printer.go

```diff
@@ -61,115 +56,72 @@ func (es Expressions) String() (s string) {
-				matching += "left"
-			} else {
-				matching += "right"
-			}
-			matching += fmt.Sprintf("(%s)", strings.Join(vm.Include, ", "))
-		}
-	}
-	return fmt.Sprintf("%s %s%s%s %s", node.LHS, node.Op, returnBool, matching, node.RHS)
+	return prettify(node)
 }
 
 func (node *Call) String() string {
-	return fmt.Sprintf("%s(%s)", node.Func.Name, node.Args)
+	return prettify(node)
 }
 
 func (node *MatrixSelector) String() string {
-	// Copy the Vector selector before changing the offset
-	vecSelector := *node.VectorSelector.(*VectorSelector)
-	offset := ""
-	if vecSelector.Offset != time.Duration(0) {
-		offset = fmt.Sprintf(" offset %s", model.Duration(vecSelector.Offset))
-	}
-
-	// Do not print the offset twice.
-	vecSelector.Offset = 0
-
-	return fmt.Sprintf("%s[%s]%s", vecSelector.String(), model.Duration(node.Range), offset)
+	return prettify(node)
 }
 
 func (node *SubqueryExpr) String() string {
-	step := ""
-	if node.Step != 0 {
-		step = model.Duration(node.Step).String()
-	}
-	offset := ""
-	if node.Offset != time.Duration(0) {
-		offset = fmt.Sprintf(" offset %s", model.Duration(node.Offset))
-	}
-	return fmt.Sprintf("%s[%s:%s]%s", node.Expr.String(), model.Duration(node.Range), step, offset)
+	return prettify(node)
 }
 
 func (node *NumberLiteral) String() string {
-	return fmt.Sprint(node.Val)
+	return prettify(node)
 }
 
 func (node *ParenExpr) String() string {
-	return fmt.Sprintf("(%s)", node.Expr)
+	return prettify(node)
 }
 
 func (node *StringLiteral) String() string {
-	return fmt.Sprintf("%q", node.Val)
+	return prettify(node)
 }
 
 func (node *UnaryExpr) String() string {
-	return fmt.Sprintf("%s%s", node.Op, node.Expr)
+	return prettify(node)
 }
 
 func (node *VectorSelector) String() string {
-	labelStrings := make([]string, 0, len(node.LabelMatchers)-1)
-	for _, matcher := range node.LabelMatchers {
-		// Only include the __name__ label if its equality matching and matches the name.
-		if matcher.Name == labels.MetricName && matcher.Type == labels.MatchEqual && matcher.Value == node.Name {
-			continue
-		}
-		labelStrings = append(labelStrings, matcher.String())
-	}
-	offset := ""
-	if node.Offset != time.Duration(0) {
-		offset = fmt.Sprintf(" offset %s", model.Duration(node.Offset))
-	}
+	return prettify(node)
+}
 
-	if len(labelStrings) == 0 {
-		return fmt.Sprintf("%s%s", node.Name, offset)
+func prettify(node Expr) string {
+	var expression string
+	switch n := node.(type) {
+	case *AggregateExpr:
+		expression = n.ExprString()
+	case *BinaryExpr:
+		expression = n.ExprString()
+	case *Call:
+		expression = n.ExprString()
+	case *MatrixSelector:
+		expression = n.ExprString()
+	case *SubqueryExpr:
+		expression = n.ExprString()
+	case *NumberLiteral:
+		expression = n.ExprString()
+	case *StringLiteral:
+		expression = n.ExprString()
+	case *VectorSelector:
+		expression = n.ExprString()
+	case *ParenExpr:
+		expression = n.ExprString()
+	case *UnaryExpr:
+		expression = n.ExprString()
+	}
```

Comment:
> We could have a single function which takes `node.ExprString()` as input

^ this

I mean have a common method like this:

```
func methodName(expr string) string {
	prettyExpr, err := PromqlPrettier(expr).Prettify()
	if err != nil {
		return expr
	}
	return prettyExpr
}
```

And call `return methodName(node.ExprString())` everywhere.